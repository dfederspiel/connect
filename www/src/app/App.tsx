import { AuthProvider } from '../context/AuthenticationContext';
import { ApolloAuthProvider } from '../context/ApolloAuthContext/ApolloAuthContext';
import { SnackBarProvider } from '../context/AlertContext/SnackBarProvider';
import AppView from './AppView';

const App = (): JSX.Element => {
  return (
    <AuthProvider>
      <ApolloAuthProvider>
        <SnackBarProvider timeout={5000}>
          <AppView />
        </SnackBarProvider>
      </ApolloAuthProvider>
    </AuthProvider>
  );
};

export default App;
