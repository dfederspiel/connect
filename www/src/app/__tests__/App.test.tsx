import { render } from '@testing-library/react';
import App from '../App';
import 'cross-fetch/polyfill';

beforeAll(() => {
  process.env.APOLLO_HOST = 'http://localhost/graphql';
  process.env.APOLLO_WS_HOST = 'ws://localhost/ws';
});

describe('the app component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
