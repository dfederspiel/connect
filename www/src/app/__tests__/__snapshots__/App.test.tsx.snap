// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app component renders correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Login
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </p>
</DocumentFragment>
`;
